format/FlightSql.proto

```diff
@@ -1779,9 +1796,60 @@ message CommandPreparedStatementUpdate {
 }
 
 /*
- * Returned from the RPC call DoPut when a CommandStatementUpdate
- * CommandPreparedStatementUpdate was in the request, containing
- * results from the update.
+ * Represents a bulk ingestion request. Used in the command member of FlightDescriptor
+ * for the the RPC call DoPut to cause the server load the contents of the stream's
+ * FlightData into the target destination.
+ */
+message CommandStatementIngest {
```

Comment:
general comment, it seems like it might be cleaner to model this as two separate operations:
1.  A DDL operation to setup the table based on various conditions.
2. The bulk ingest operations.

I also worry about the slippery slope here of other options (e.g. when the schema of the bulk data doesn't match the existing schema).

Reply:
I think splitting into separate 'create table' and 'append table' operations is reasonable for the protocol level here.

Absent a transaction, the difference is that there's now the possibility of a race condition with another writer. But in such cases you would want an explicit transaction anyways.

Reply:
I've given some thought to potentially splitting up the "create" and "append" operations. Having creation be a separate operation would be nice, but I do also see the slippery slope of adding more configuration over time for various edge cases.

It would definitely be valuable to have _some_ method of abstracting table creation logic in a way that's independent of the backend. I did some research and learned that Substrait already has a [relation](https://substrait.io/relations/logical_relations/#ddl-data-definition-language-operator) defined for this. For this reason, it seems that independent "create" and "append" operations are more aligned with Substrait's stated goals rather than those of Flight SQL. I think that past a certain level of detail, a Substrait DDL + Write plan should be the supported/preferred approach to handle these kinds of scenarios.